genrl/agents/modelbased/base.py

```diff
@@ -0,0 +1,65 @@
+from abc import ABC
+
+import torch
+
+
+class Planner:
+    def __init__(self, initial_state, dynamics_model=None):
+        if dynamics_model is not None:
+            self.dynamics_model = dynamics_model
+        self.initial_state = initial_state
+
+    def _learn_dynamics_model(self, state):
+        raise NotImplementedError
+
+    def plan(self):
+        raise NotImplementedError
+
+    def execute_actions(self):
+        raise NotImplementedError
+
+
+class ModelBasedAgent(ABC):
```

Comment:
Can this inherit from the genrl/deep `BaseAgent`?

Reply:
It can, and I think thats a better option (for now at least)